omod/src/main/java/org/openmrs/module/fhir/providers/RestfulLocationResourceProvider.java

```diff
@@ -109,33 +110,68 @@ public void deleteLocation(@IdParam IdDt theId) {
 	}
 	
 	/**
-	 * Update location by id.
-	 *
+	 * Update Location
+	 * 
+	 * @param location fhir Location object
+	 * @param theId , the uuid of the Location resource to be update
+	 * @return Method outcome contains the status of the update operation
+	 */
+	@Update
+	public MethodOutcome updateLocation(@ResourceParam Location location, @IdParam IdDt theId) {
+		MethodOutcome retVal = new MethodOutcome();
+		OperationOutcome outcome = new OperationOutcome();
+		location = locationResource.updateLocation(theId.getIdPart(), location);
+		outcome.addIssue().setDetails("Location successfully updated");
+		retVal.setOperationOutcome(outcome);
+		return retVal;
+	}
+
+	/**
+	 * Conditionally update location by name.
 	 * @param theLocation {@link ca.uhn.fhir.model.dstu2.resource.Location} object provided by the
 	 *            {@link ca.uhn.fhir .rest.server.RestfulServer}
 	 * @param theId Only one of theId or theConditional will have a value and the other will be
 	 *            null, depending on the URL passed into the server
-	 * @param theConditional This will have a value like "Patient?identifier=system%7C00001
-	 * @return This object contains the identity of the created resource.
+	 * @param theConditional This will have a value like "Location?name=Colombo
+	 * @return MethodOutcome which contains the status of the operation
 	 */
 	@Update()
-	public MethodOutcome updateLocationById(@ResourceParam Location theLocation, @IdParam IdDt theId,
+	public MethodOutcome updateLocationByName(@ResourceParam Location theLocation, @IdParam IdDt theId,
 	                                        @ConditionalUrlParam String theConditional) {
-		MethodOutcome methodOutcome = new MethodOutcome();
-		String id = null;
+		MethodOutcome outcome = new MethodOutcome();
+		OperationOutcome operationoutcome = null;
 		if (theConditional != null) {
-			int startIndex = theConditional.lastIndexOf('=');
-			id = theConditional.substring(startIndex + 1);
+			List<Location> locationList = null;
+			String locationName = null;
+			try {
+				int startIndex = theConditional.lastIndexOf('=');
+				locationName = theConditional.substring(startIndex + 1);
+			}
+			catch (NullPointerException e) {
+				operationoutcome = new OperationOutcome();
```

Comment:
@harsha89  @surangak  i changed the openmrs formatter.

Window >> Preferences >> Java >> Code Style >> Formatter in the Control Statements tab:
Insert a new line before 'catch' in a 'try' statement
Insert a new line before 'finally' in a 'try' statement

Now the try catch new line is fixed. But why are having such wrong style in openmrs formatter?

There are several placed in fhir code base where the try catch spaces are not properly done. Do we need to change all lines? 

Editer>save options>edit all lines will work for this.